source/material_model/reaction_model/katz2003_mantle_melting.cc

```diff
@@ -150,6 +171,196 @@ namespace aspect
+                      {
+                        if (c == peridotite_idx && this->get_timestep_number() > 0)
+                          reaction_rate_out->reaction_rates[i][c] = porosity_change / melting_time_scale * (1 - maximum_melt_fraction) / (1 - old_porosity);
+                        else if (c == porosity_idx && this->get_timestep_number() > 0)
+                          reaction_rate_out->reaction_rates[i][c] = porosity_change / melting_time_scale;
+                        else
+                          reaction_rate_out->reaction_rates[i][c] = 0.0;
+                      }
+                    out.reaction_terms[i][c] = 0.0;
+                  }
+
+                const double porosity = std::min(1.0, std::max(in.composition[i][porosity_idx],0.0));
+                out.viscosities[i] *= exp(- alpha_phi * porosity);
+
+                out.entropy_derivative_pressure[i]    = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), maximum_melt_fraction, NonlinearDependence::pressure);
+                out.entropy_derivative_temperature[i] = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), maximum_melt_fraction, NonlinearDependence::temperature);
+              }
+            else
+              {
+                out.entropy_derivative_pressure[i]    = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), 0, NonlinearDependence::pressure);
+                out.entropy_derivative_temperature[i] = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), 0, NonlinearDependence::temperature);
+
+                // no melting/freezing is used in the model --> set all reactions to zero
+                for (unsigned int c=0; c<in.composition[i].size(); ++c)
+                  {
+                    out.reaction_terms[i][c] = 0.0;
+
+                    if (reaction_rate_out != nullptr)
+                      reaction_rate_out->reaction_rates[i][c] = 0.0;
+                  }
+              }
+          }
+
+      }
+
+      template <int dim>
+      void
+      Katz2003MantleMelting<dim>::
+      calculate_fluid_outputs(const typename Interface<dim>::MaterialModelInputs &in,
+                              typename Interface<dim>::MaterialModelOutputs &out,
+                              const double thermal_expansivity,
+                              const double reference_T) const
+      {
+        MeltOutputs<dim> *melt_out = out.template get_additional_output<MeltOutputs<dim>>();
+
+        if (melt_out != nullptr)
+          {
+            const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
+            for (unsigned int i=0; i<in.n_evaluation_points(); ++i)
+              {
+                double porosity = std::max(in.composition[i][porosity_idx],0.0);
+
+                melt_out->fluid_viscosities[i] = eta_f;
```

Comment:
Per the comment above, I think I would be good to rename this variable to `viscosity_fluid`, `viscosity_fluid`, `melt_viscosity`, etc.